### PR TITLE
generic: sycl: deconvolution: implemented

### PR DESCRIPTION
**Pull request by t4c1**
Implemented SYCL deconvolution. For fwd and bwd data it uses existing implementations. For bwd_weights it uses existing convolution kernel, which is modified a bit.

### CODE REVIEW

**Comment by t4c1** [comment]
make test
enable device_gpu
enable thr_sycl
enable thr_cuda

**Comment by densamoilov** [comment]
@t4c1, minor reminder, can you please use `gpu: generic: sycl` instead of `gpu: sycl` in commit messages?

**Comment by densamoilov** [comment]
@t4c1, also, we are waiting for Codeplay's review of this PR.

**Review by kala855 — APPROVED** [review]

**Comment by t4c1** [comment]
make test
disable device_cpu
disable thr_ocl
disable thr_omp
disable thr_tbb_oneapi
enable thr_cuda
enable thr_hip
enable benchdnn_all
enable arch_rtx
enable arch_instinct
disable os_mac_cpu

**Review by sgeor255 — APPROVED** [review]

File: src/gpu/generic/sycl/ref_deconvolution.hpp
Comment (t4c1) [reply src/gpu/generic/sycl/ref_deconvolution.hpp:65]: done

File: src/gpu/generic/sycl/ref_deconvolution.hpp
Comment (t4c1) [reply src/gpu/generic/sycl/ref_deconvolution.hpp:82]: Added a newline to make it clearer.

**Comment by t4c1** [comment]
make test
enable device_gpu
disable device_cpu
disable thr_ocl
disable thr_omp
disable thr_tbb_oneapi
enable thr_cuda
enable thr_hip
enable benchdnn_all
enable arch_rtx
enable arch_instinct
disable os_mac_cpu

File: src/gpu/generic/sycl/ref_deconvolution.hpp
Comment (mgouicem) [inline src/gpu/generic/sycl/ref_deconvolution.hpp:114]: In general, scales are used for inference.
Hence scales are typically not supported for bwd by weights.

File: src/gpu/generic/sycl/ref_deconvolution.cpp
Comment (mgouicem) [inline src/gpu/generic/sycl/ref_deconvolution.cpp:86]: I guess this could be simplified, as bwd by weights typically does not support quantization (so no scales/zero-point).

Comment (t4c1) [reply src/gpu/generic/sycl/ref_deconvolution.cpp]: removed

**Comment by t4c1** [comment]
make test
enable device_gpu
enable thr_cuda

**Review by mgouicem — APPROVED** [review]
Thanks. Please remember to squash your commits before merging.

**Comment by vpirogov** [comment]
make test
enable device_gpu
enable thr_cuda

**Comment by vpirogov** [comment]
@t4c1, NVIDIA validation started successfully.

**Comment by vpirogov** [comment]
@t4c1, there are few more things to address on this PR:
* Merge commits are not allowed in main branch. Please rebase your changes.
* Commits messages should be formatted according to [code contribution guidelines](https://github.com/oneapi-src/oneDNN/blob/main/CONTRIBUTING.md#code-contribution-guidelines)
* Address failed ClangFormat check

**Comment by t4c1** [comment]
Thanks for the reminder @vpirogov, I missed those. 

Also, I just made a mistake rebasing and forced-pushed different set of changes than intended. It should be fixed now, but github automatically requested reviews from various teams. Can those requests be removed?

**Pull request merged**
